core/src/socket.rs

```diff
@@ -111,100 +160,229 @@ impl<'gc> Sockets<'gc> {
     }
 
     pub fn close(&mut self, handle: SocketHandle) {
-        if let Some(Socket { sender, .. }) = self.sockets.remove(handle) {
+        if let Some(Socket { sender, target }) = self.sockets.remove(handle) {
             drop(sender); // NOTE: By dropping the sender, the reading task will close automatically.
+
+            // Clear the buffers if the connection was closed.
+            match target {
+                SocketKind::Avm1(target) => {
+                    let target =
+                        XmlSocket::cast(target.into()).expect("target should be XmlSocket");
+
+                    target.read_buffer().clear();
+                }
+                SocketKind::Avm2(target) => {
+                    target.read_buffer().clear();
+                    target.write_buffer().clear();
+                }
```

Comment:
It first I thought `target.write_buffer().clear()` was missing, but it's actually on purpose as the write buffer isn't used by the Avm1 XML socket. Is it worth trying to move this buffer to the Avm2-only code? Or to add a comment to explain the purpose of the writer buffer? I don't know if it's how much Ruffle team cares about that

Reply:
The reason that Avm2 Socket has a write buffer is because generally data is not sent out until the flush() method is called and at that point we empty out the buffer and send it out. But with XMLSocket we send the data out immediately, so there is no need to buffer the data.